src/components/CreateCoupons/index.jsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from "react";
+import "./style.css";
+import { Typography, Button, Row, Col, Avatar, message } from "antd";
+import { InfoCircleOutlined } from "@ant-design/icons";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const db = firebase.firestore();
+const timestamp = firebase.firestore.FieldValue.serverTimestamp;
+const { Title, Text } = Typography;
+
+const CreateCoupons = ({ isTesting }) => {
+  const { t } = useTranslation();
+  const [count, setCount] = useState(1);
+  const [newCoupons, setNewCoupons] = useState({
+    compName: "",
+    amounts: {
+      25: false,
+      50: false,
+      75: false,
+      100: false,
+      150: false,
+      200: false,
+    },
+  });
+  const [ghosted, setGhosted] = useState({
+    "25₺": true,
+    "50₺": true,
+    "75₺": true,
+    "100₺": true,
+    "150₺": true,
+    "200₺": true,
+  });
+  const postCoupons = () => {
+    // db.collection("restaurants").where("uid", "==", "")
+    db.collection("coupons")
+      .add({
+        compName: newCoupons.compName,
+        amounts: { ...newCoupons.amounts },
+        createdAt: timestamp(),
+        img:
+          "https://image.freepik.com/free-photo/wooden-board-empty-table-top-blurred-background_1253-1584.jpg",
+      })
+      .then((docRef) =>
+        db.collection("coupons").doc(docRef.id).update({
+          id: docRef.id,
+        })
+      )
+      .then(() => message.success(t("createCoupons.message.success")))
+      .catch(() => message.warning(t("createCoupons.message.warning")));
+  };
+  const selectCoupon = (e) => {
+    const key = e.target.innerText;
+    const slicedKey = key.slice(0, key.length - 1);
+    setGhosted({
+      ...ghosted,
+      [key]: !ghosted[key],
+    });
+
+    setNewCoupons({
+      ...newCoupons,
+      compName: "Arby's",
+      amounts: {
+        ...newCoupons.amounts,
+        [slicedKey]: !newCoupons.amounts[slicedKey],
+      },
+    });
+
+    if (ghosted[key] === false) {
+      setCount((previous) => previous - 1);
+    } else {
+      setCount((previous) => previous + 1);
+    }
+  };
+  const handleSubmit = () => {
+    postCoupons();
+    setGhosted({
+      "25₺": true,
+      "50₺": true,
+      "75₺": true,
+      "100₺": true,
+      "150₺": true,
+      "200₺": true,
+    });
+    setCount(1);
+  };
+
+  return (
+    <div className="coupons">
+      <Title className="createCouponsTitle" level={3} type="primary">
+        {t("createCoupons.title")}
+      </Title>
+      <Row gutter={[16, 16]}>
+        <Col span={8}>
```

Comment:
from line 94 till line 110 is repeated alot of times, please use map for this one